src/middlewared/middlewared/plugins/cron.py

```diff
@@ -195,3 +198,68 @@ def cron_extend(self, data):
         await self.middleware.call('service.restart', 'cron')
 
         return response
+
+    @accepts(
+        Int('id')
+    )
+    @job()
+    async def run(self, job, id):
+        """
+        Job to run cronjob task of `id`.
+        """
+        cron_task = await self._get_instance(id)
+        cron_cmd = ' '.join(
+            (await self.construct_cron_command(
+                cron_task['schedule'], cron_task['user'],
+                cron_task['command'], cron_task['stdout'], cron_task['stderr']
+            ))[6:]
+        )
+
+        job.set_progress(
+            10,
+            'Executing Cron Task'
+        )
+
+        p = await Popen(
+            cron_cmd, stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = await p.communicate()
```

Comment:
You're ignoring the whole block of running the cron with the specified user and setting context.

See how it was done for rsync, we need to not duplicate code while doing it.